cli/mender-artifact/copy.go

```diff
@@ -117,6 +118,48 @@ func doCopy(c *cli.Context, repack *bool, r *io.ReadCloser,
 	return nil
 }
 
+// Install installs a file from the host filesystem onto either
+// a mender artifact, or an sdimg.
+func Install(c *cli.Context) (err error) {
+	fmt.Println("Installing...")
+	var r io.ReadCloser
+	var w PartitionReadWriteClosePacker
+	switch parseCLIOptions(c) {
+	case copyin:
+		var perm os.FileMode
+		if c.String("mode") == "" {
+			return cli.NewExitError("File permissions needs to be set, if you are simply copying, the cp command should fit your needs", 1)
+		}
+		var i int
+		i, err = strconv.Atoi(c.String("mode"))
+		if err != nil {
+			return cli.NewExitError(fmt.Sprintf("%v", err), 1)
+		}
+		perm = os.FileMode(i)
+		r, err = os.OpenFile(c.Args().First(), os.O_RDWR, perm)
```

Comment:
I'm confused. Why are you using `perm` on a file you intend to read? It has no effect. And mode should be `O_RDONLY` for a file you will only read.

Reply:
The permission bits are copied over from debugfs